lib/libvmod_blob/vmod_blob.c

```diff
@@ -246,54 +222,61 @@ vmod_blob_get(VRT_CTX, struct vmod_blob_blob *b)
 
 VCL_STRING v_matchproto_(td_blob_blob_encode)
 vmod_blob_encode(VRT_CTX, struct vmod_blob_blob *b, VCL_ENUM encs,
-		 VCL_ENUM case_s)
+    VCL_ENUM case_s)
 {
-	enum encoding enc = parse_encoding(encs);
-	AENC(enc);
+	const struct vmod_blob_codec *enc;
 	enum case_e kase = parse_case(case_s);
+	ssize_t len;
+	int idx = -1;
+	char *s;
 
 	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
 	CHECK_OBJ_NOTNULL(b, VMOD_BLOB_MAGIC);
+	AN(encs);
+	AN(case_s);
 
-	if (!check_enc_case(ctx, encs, case_s, enc, kase))
-		return NULL;
+	enc = parse_encoding(encs, &idx);
+	AN(enc);
+	assert(idx >= 0);
+
+	if (!check_enc_case(ctx, enc, case_s, kase))
+		return (NULL);
 	if (b->blob.len == 0)
-		return "";
+		return ("");
 	if (kase == DEFAULT)
 		kase = LOWER;
 
-	if (b->encoding[enc][kase] == NULL) {
-		AZ(pthread_mutex_lock(&b->lock));
-		if (b->encoding[enc][kase] == NULL) {
-			ssize_t len = func[enc].encode_l(b->blob.len);
+	if (b->encoding[idx][kase] != NULL)
```

Comment:
Checking != NULL outside of the lock, and avoiding the double-check under the lock, is a good move.